react_juce/core/AppHarness.cpp

```diff
@@ -18,61 +18,43 @@ namespace reactjuce
         : appRoot(_appRoot)
     {
         JUCE_ASSERT_MESSAGE_THREAD
-
-        fileWatcher = std::make_unique<FileWatcher>([this]() {
```

Comment:
@nick-thompson , I think using this single implementation of `handleFilesChanged` is OK? Let me know if I've aggressively blown over some edge case here which I've forgotten or not thought about. This is nice if we can keep it I think. Removes some duplication and keeps the bytecode facility fairl well encapsulated. I prefer this to a dynamic_cast.